include/sentry.h

```diff
@@ -338,26 +338,86 @@ typedef enum sentry_level_e {
 } sentry_level_t;
 
 /**
- * Creates a new empty event value.
+ * Creates a new empty Event value.
+ *
+ * See https://develop.sentry.dev/sdk/event-payloads/
  */
 SENTRY_API sentry_value_t sentry_value_new_event(void);
 
 /**
- * Creates a new message event value.
+ * Creates a new Message Event value.
+ *
+ * See https://develop.sentry.dev/sdk/event-payloads/message/
  *
  * `logger` can be NULL to omit the logger value.
  */
 SENTRY_API sentry_value_t sentry_value_new_message_event(
     sentry_level_t level, const char *logger, const char *text);
 
 /**
- * Creates a new breadcrumb with a specific type and message.
+ * Creates a new Breadcrumb with a specific type and message.
+ *
+ * See https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
  *
  * Either parameter can be NULL in which case no such attributes is created.
  */
 SENTRY_API sentry_value_t sentry_value_new_breadcrumb(
     const char *type, const char *message);
 
+/**
+ * Creates a new Exception value.
+ *
+ * See https://develop.sentry.dev/sdk/event-payloads/exception/
+ *
+ * The returned value needs to be attached to an event via
+ * `sentry_event_value_add_exception`.
+ */
+SENTRY_EXPERIMENTAL_API sentry_value_t sentry_value_new_exception(
+    const char *type, const char *value);
```

Comment:
What are the nullability-constraints on this API? Please add this to the doc comment.

The implementation uses `sentry_value_new_string`, which returns `sentry_value_new_null()` if the pointer is NULL, which is good behavior in my opinion. In `sentry_value_new_thread` you even make an explicit check for name. 

Note that _one of_ `type` or `value` are required for valid exceptions by the server.

Reply:
hm, https://develop.sentry.dev/sdk/event-payloads/exception/#attributes does not really talk about "*one of* them being required". I read it that both are, and that is what the API reflects.

Reply:
Thanks, we need to update that. The actual behavior is:

https://github.com/getsentry/relay/blob/10f2c93efb8b5a200168a8fb961c3b136c1e5f18/relay-general/src/store/normalize.rs#L618-L623

Reply:
Update: In the long run we will need two functions, one for creating synthetic errors and one for language-level exceptions. Since this function is for language-level exceptions, it's better to require both the type and the value.